docs/content/en/docs/workflows/dev.md

```diff
@@ -4,4 +4,44 @@ linkTitle: "Continuous development"
 weight: 1
 ---
 
-{{< todo 3074>}} dev page
\ No newline at end of file
+## `skaffold dev`
+
+`skaffold dev` enables continuous local development on an application.
+While in `dev` mode, Skaffold will watch an application's source files, and when it detects changes,
+will rebuild your images (or sync files to your running containers), push any new images, and redeploy the application to your cluster.
+
+`skaffold dev` is considered Skaffold's main mode of operation, as it allows you
+to leverage all of the main features of Skaffold in a continuous way while iterating
+on your application.
+
+
+## Dev loop
+
+When `skaffold dev` is run, Skaffold will first do a full build and deploy of all artifacts specified in the `skaffold.yaml`, identical behavior to `skaffold run`. Upon successful build and deploy, Skaffold will start watching all source file dependencies for all artifacts specified in the project. As changes are made to these source files, Skaffold will rebuild the associated artifacts, and redeploy the new changes to your cluster.
+
+The dev loop will run until the user cancels the Skaffold process with `Ctrl+C`. Upon receiving this signal, Skaffold will clean up all deployed artifacts on the active cluster, meaning that Skaffold won't abandon any Kubernetes resources that it created throughout the lifecycle of the run.
+
+
+## Precedence of Actions
+
+The actions performed by Skaffold during the dev loop have precendence over one another, so that behavior is always predictable. The order of actions is:
+
+1) File Sync
+1) Build
+1) Deploy
+
+
+## File Watcher and Watch Modes
+
+Skaffold computes the dependencies for each artifact based on the builder being used, and the root directory of the artifact. Once all source file dependencies are computed, in `dev` mode, Skaffold will continuously watch these files for changes in the background, and conditionally re-run the loop when changes are detected.
+
+By default, Skaffold uses `fsnotify` to monitor events on the local filesystem. Skaffold also supports a `polling` mode where the filesystem is checked for changes on a configurable interval, or a `manual` mode, where Skaffold waits for user input to check for file changes. These watch modes can be configured through the `--trigger` flag.
+
+
+## Control API
```

Comment:
I think we should explain trigger types here and then control api.

Reply:
also, I'm writing a section about the Control API :)

Reply:
I just wanted to have a little section about it here for discoverability, and then link in the full docs to the control API. I mentioned the trigger types above this section. do you think I should remove this?

Reply:
I'd remove it, it is an advanced topic for those who want to integrate tools with `skaffold dev` - it is not essential for the CLI users. WDYT?

Reply:
I agree, control API is not for CLI user and more of integration for tools.

Reply:
sounds reasonable, I do think CLI users could still potentially make use of the control API, but it's not a first class feature. I cut down the blurb to just explain what it is and why you might use it, and then pointed to the API docs